bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -698,6 +698,13 @@ public AnalysisResultAdded newAnalysisResult(BlackboardArtifact.Type artifactTyp
 			throw new TskCoreException("Error adding analysis result.", ex);
 		}
 	}
+	
+	@Override
+	public DataArtifact newDataArtifact(BlackboardArtifact.Type artifactType, Collection<BlackboardAttribute> attributesList, OsAccount osAccount) throws TskCoreException {
+
+		return sleuthkitCase.getBlackboard().newDataArtifact(artifactType, this.getObjectID(), this.getDataSource().getId(), attributesList, osAccount);
```

Comment:
The current behavior is that we dont' make BlackboardArtifacts as children of BlackboardArtifacts.  I'd propose we'd have similar behavior for DataArtifacts, which means this should throw an exception.

Reply:
Addressed.